hnn_core/gui/gui.py

```diff
@@ -1330,23 +1340,23 @@ def on_upload_params_change(change, params, tstop, dt, log_out, drive_boxes,
 
     ext = Path(params_fname).suffix
     read_func = {'.json': _read_json, '.param': _read_legacy_params}
-    params_network = read_func[ext](param_data)
+    params_network = Params(read_func[ext](param_data))
```

Comment:
This is the fix ... after reading the params from the file, it is harmonized with some expected defaults and this is done using the `Params` class. When we initially load the default param, we load it using the `read_params` function. However, because of how ipywidgets handles file reading, I cannot use `read_params` directly ... it gives me the file contents as a string and it has to be parsed using the private functions `_read_json` or `_read_legacy_params`.